lib/utils.js

```diff
@@ -55,8 +55,24 @@ function stripHttp1ConnectionHeaders (headers) {
   return dest
 }
 
+function filterHeaders (headers, filter) {
```

Comment:
Do you see any other uses for this function? I am just afraid of adding an extra computational effort just to delete the `content-length` header.